src/lib/define-dynamic-block.js

```diff
@@ -176,14 +219,64 @@ const defineDynamicBlock = (ScratchBlocks, categoryInfo, staticBlockInfo, extend
                 }
                 return `%${++argCount}`;
             });
+            const wasRendered = this.rendered;
+            this.rendered = false;
+            const connectionMap = this.disconnectOldBlocks_();
```

Comment:
Keep track of blocks (and shadows) that were previously connected to this block before we start dynamically updating its layout, so that we can reconnect them after we're done.

Reply:
Does `connectionMap` also contain the information destroyed by `removeAllInputs_()`? Or is that what the stuff after `interpolate_()` is doing?

Reply:
`removeAllInputs_` actually just clears all the input holes from the block. It has nothing to do with previously connected shadows etc. (`disconnectOldBlocks_` takes care of that)